### ISSUE TITLE
Training Custom Dataset

### ISSUE THREAD

**Issue by samin9796** [espnet/espnet#2314]
Hi! I want to train a model on my own dataset using docker container (ASR). I have got the following questions:

1. I came to know that I need to prepare text, wav.scp, utt2spk and spk2utt files. But I don't have speaker information. Is there any way to work without the last two files?
2. How can I initiate training? Do I need to split the dataset into train, test and validation sets? If so, how to proceed after that? How can I set the directory for my own dataset?
3. Do I have to build a Dictionary and do JSON format data preparation?

Thank you!

**Comment by kamo-naoyuki** [comment]
I recommend you using espnet2 . More easy.

> 1. I came to know that I need to prepare text, wav.scp, utt2spk and spk2utt files. But I don't have speaker information. Is there any way to work without the last two files?

Speaker information is not used now for ASR at all, except for the definition of validation set, therefore utt2spk and spk2utt can be dummy.
See: https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE#about-kaldi-style-data-directory

> 2. How can I initiate training? Do I need to split the dataset into train, test and validation sets? If so, how to proceed after that? How can I set the directory for my own dataset?

You need to prepare a validation set at least. If it's not defined, normally, it's 1-5% samples of the training set randomly selected.

> 3. Do I have to build a Dictionary and do JSON format data preparation?

Use espnet2.

**Comment by samin9796** [comment]
@kamo-naoyuki 

> ./run.sh --docker-gpu 0 --is-egs2 --docker-egs an4/asr1 --ngpu1

For my own corpus, do I need to modify this part- "--docker-egs an4/asr1" ?

**Comment by kamo-naoyuki** [comment]
I don't know about docker. @Fhrozen

**Comment by Fhrozen** [comment]
@samin9796 For using docker, you will need to add a new dir on egs2 with the name of your recipe and the task;
so:
```
<espnet_root>
       |---- egs2
               |-----my_recipe(an4)
                         |------my_task(asr1)
```
So yes, you can run with `./run.sh --docker-gpu 0 --is-egs2 --docker-egs <my_new_recipe>/<my_task> --ngpu1`, change the recipe & task dirname

**Comment by samin9796** [comment]
@Fhrozen  I downloaded the docker image. I don't have anything in the `<my_new_recipe>/asr1` directory (No run.sh file). Could you please help me what to do next?

**Comment by kamo-naoyuki** [comment]
Please read https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE and documentation all before asking.
